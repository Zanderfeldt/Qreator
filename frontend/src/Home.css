/* Home.css */

body {
  margin: 0;
  padding: 0;
}

.home-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.Home {
  text-align: center;
  flex-direction: column; 
  margin: 120px auto;
}

.Home h1 {
  font-size: 3rem;
  font-weight: bold;
  margin-top: 2px;
}

.Home p {
  font-size: 1.2rem;
  margin-bottom: 20px;
}

.Home-buttons {
  display: flex;
  justify-content: center; 
  margin-top: 20px;
}

.Home-buttons a {
  margin: 0 4px;
  padding: 6px 10px;
  font-size: 0.8rem;
  text-decoration: none;
  color: #fff;
  background-color: #62865d;
  border-radius: 1px;
}

