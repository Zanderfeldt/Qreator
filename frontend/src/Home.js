import React, { useContext } from "react";
import "./Home.css";
import UserContext from "./auth/UserContext";
import { Link } from "react-router-dom";
import logo from "./codeIcon.png";

function Home() {
  const { currUser } = useContext(UserContext);

  return (
    <div className="Home">
      <img src={logo} alt="logo" style={{width: "150px"}}></img>
      <h1>QReator</h1>
      <p>Design your own QR Codes in one, convenient place.</p>
      { currUser ?
      <div> 
      <h2>
        Welcome Back, {currUser.firstName || currUser.username}! 
      </h2>
      <div className="Home-buttons">
      <Link to="/create" className="Home-buttons signup">Start Qreating</Link>
      </div>
      </div>
      : (<div className="Home-buttons">
      <Link to="/signup" className="Home-buttons signup">Sign Up</Link>
      <Link to="/login" className="Home-buttons login">Log In</Link>
    </div>
    )}
      
    </div>
  )

}

export default Home;