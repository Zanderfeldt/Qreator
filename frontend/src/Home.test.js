import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';
import UserContext from './auth/UserContext';
import { MemoryRouter, BrowserRouter } from 'react-router-dom'; 

//smoke test
test("renders without crashing", function() {

  const contextValue = {
    currUser: {
      firstName: 'test',
      lastName: 'test',
      username: 'test',
      email: 'test@test.com'
    }
  };
  const { container } = render(
    <MemoryRouter>
    <UserContext.Provider value={contextValue}>
      <Home />
    </UserContext.Provider>
    </MemoryRouter>
  );
  expect(container).toBeDefined(); //Verify Home component renders correctly
  expect(screen.getByText('Design your own QR Codes in one, convenient place.')).toBeInTheDocument(); //Verify Welcome message appears on Home page

    //Verify 'Start Qreating' button on Home Page
  const createButton = screen.getByText('Start Qreating'); 
  expect(createButton).toBeInTheDocument();

});


//snapshot test
test('Home component matches snapshot', () => {
  const mockUser = { currUser: 'Test User' }; // Define a mock user

  const { container } = render(
    <UserContext.Provider value={mockUser}>
      <BrowserRouter>
       <Home />
      </BrowserRouter>
    </UserContext.Provider>
  );

  expect(container).toMatchSnapshot();
});