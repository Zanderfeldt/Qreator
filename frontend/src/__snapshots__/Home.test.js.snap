// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div>
  <div
    class="Home"
  >
    <img
      alt="logo"
      src="codeIcon.png"
      style="width: 150px;"
    />
    <h1>
      QReator
    </h1>
    <p>
      Design your own QR Codes in one, convenient place.
    </p>
    <div>
      <h2>
        Welcome Back, 
        !
      </h2>
      <div
        class="Home-buttons"
      >
        <a
          class="Home-buttons signup"
          href="/create"
        >
          Start Qreating
        </a>
      </div>
    </div>
  </div>
</div>
`;
